mongolog/handlers.py

```diff
@@ -231,14 +154,40 @@ def check_keys(self, record):
             return record
 
         for k, v in record['msg'].items():
-            record['msg'][self.new_key(k)] = record['msg'].pop(k)
-
-            if isinstance(v, dict):
-                for old_key in record['msg'][k].keys():
-                    record['msg'][k][self.new_key(old_key)] = record['msg'][k].pop(old_key)
+            self._check_keys(k, v, record['msg'])
 
         return record
 
+    def _check_keys(self, k, v, _dict):
```

Comment:
This is now a recursive check that goes as deep as required to check all key's.  Previously it only went 2 levels deep.